presto-iceberg/src/main/java/com/facebook/presto/iceberg/MetadataLogTable.java

```diff
@@ -0,0 +1,122 @@
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadata.MetadataLogEntry;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.DateTimeEncoding.packDateTimeWithZone;
+import static com.facebook.presto.common.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class MetadataLogTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = ImmutableList.<ColumnMetadata>builder()
+            .add(new ColumnMetadata("timestamp", TIMESTAMP_WITH_TIME_ZONE))
+            .add(new ColumnMetadata("file", VARCHAR))
+            .add(new ColumnMetadata("latest_snapshot_id", BIGINT))
+            .add(new ColumnMetadata("latest_schema_id", BIGINT))
+            .add(new ColumnMetadata("latest_sequence_number", BIGINT))
+            .build();
+
+    public MetadataLogTable(SchemaTableName tableName, Table icebergTable)
+    {
+        tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"), COLUMNS);
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        InMemoryRecordSet.Builder table = InMemoryRecordSet.builder(COLUMNS);
+
+        TableMetadata metadata = ((org.apache.iceberg.BaseTable) icebergTable).operations().current();
+        List<MetadataLogEntry> metadataLogEntries = metadata.previousFiles();
+
+        processMetadataLogEntries(table, session, metadataLogEntries);
+        addLatestMetadataEntry(table, session, metadata);
+
+        return table.build().cursor();
+    }
+
+    private void processMetadataLogEntries(InMemoryRecordSet.Builder table, ConnectorSession session, List<MetadataLogEntry> metadataLogEntries)
+    {
+        if (metadataLogEntries.isEmpty()) {
+            return;
+        }
+
+        for (MetadataLogEntry entry : metadataLogEntries) {
+            Long snapshotId = null;
+            Snapshot snapshot = null;
+            try {
+                snapshotId = SnapshotUtil.snapshotIdAsOfTime(icebergTable, entry.timestampMillis());
```

Comment:
```suggestion
                snapshotId = snapshotIdAsOfTime(icebergTable, entry.timestampMillis());
```

nit: I know this code is from iceberg lib, but we can still use static import as much as possible.